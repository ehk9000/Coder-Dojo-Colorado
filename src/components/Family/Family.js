import React, { useState, useEffect } from 'react';
import { Link, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import requests from '../../utils/requests/requests';
import StudentPreview from '../StudentPreview/StudentPreview';

export const Family = ({ user, loading }) => {
  const { username } = user;
  const [students, setStudents] = useState([]);
  const [contactInfo, setContactInfo] = useState([]);
  const [redirect, setRedirect] = useState(false);

  useEffect(() => {
    if (user.role === 1) {
      const getFamily = async () => {
        const response = await requests.getFamily();

        setStudents(response.me.students);
      };

      const getContactInfo = async () => {
        const response = await requests.getGuardianData();
        const { addresses, email, phoneNumber } = response.me;
        const { 
          street1,
          street2,
          city,
          state,
          zip
        } = addresses[0];
        const info = {
          email,
          phoneNumber,
          street1,
          street2,
          zip,
          city,
          state
        };

        setContactInfo(info);
      };
      
      getFamily();

      getContactInfo();
    }
  }, [user]);

  useEffect(() => {
    if (user.role !== 1) {
      setRedirect(true);
    }
  }, [user]);

  const generateStudents = () => {
    if (students) {
      students.map(student => <StudentPreview key={student.id} student={student} />);
    }
  };

  const generateContactInfo = () => {
    const { 
      phoneNumber,
      email,
      street1,
      street2,
      city,
      state,
      zip 
    } = contactInfo;
    const address1 = `${street1}, ${street2 || ''}`;
    const address2 = `${city}, ${state} ${zip}`;

    return (
      <div className="family-sidebar">
        <h3>My Contact Info</h3>
        <h4>
          Phone:&nbsp;
        </h4>
        <p>
          {phoneNumber}
        </p>
        <h4>
          Email:&nbsp;
        </h4>
        <p>
          {email}
        </p>
        <h4>
          Address:&nbsp;
        </h4>
        <p>
          {address1}
        </p>
        <p>
          {address2}
        </p>
      </div>
    );
  };

  if (redirect) {
    return <Redirect to="/" />;
  }

  return (
    <section className="Family">
      <div className="family-header">
        <h2>{`Welcome, ${username}!`}</h2>
      </div>
      <div className="family-main">
        {generateContactInfo()}
        <div className="family-details">
          <h3 className="students-header">
            {loading ? 'Loading...' : 'My Family'}
            <Link to="/myfamily/registerstudent">
              <button type="button">Add a Student</button>
            </Link>
          </h3>
          <div className="students-container">
            {generateStudents()}
          </div>
        </div>
      </div>
    </section>
  );
};

export const mapStateToProps = state => ({
  user: state.user,
  loading: state.isFetching
});

export default connect(mapStateToProps)(Family);

Family.propTypes = {
  user: PropTypes.object,
  loading: PropTypes.bool
};

Family.defaultProps = {
  user: {},
  loading: false
};
