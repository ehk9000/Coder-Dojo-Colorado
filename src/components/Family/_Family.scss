.Family {
  width: 80%;
  margin: auto;
  height: 100vh;

  h2 {
    margin: 0 20px;
  }

  h3 {
    margin: 0 0 10px 0;
  }

  h4 {
    margin: 10px 0 0 0;
  }

  .family-header {
    width: 100%;
    padding: 30px 0;
    display: flex;
    align-items: center;
    background-color: #724695;
    color: #fff;
  }

  .family-main {
    display: flex;
    background-color: #E5E5E5;
    height: 100%;

    .family-sidebar {
      flex: 1;
      padding: 20px;
    }

    .family-details {
      background: #FFFFFF;
      flex: 2;
      padding: 20px;

      .students-header {
        display: flex;
        align-items: center;
        justify-content: space-between;
      }

      button {
        background: $footer-color;
        border: none;
        color: white;
        font-size: .8em;
        padding: 10px;
      }
    }

    .students-container {
      min-height: 1.5em;

      .loading {
        font-size: 1.5em;
        font-weight: 800;
      }
    }
  }
}