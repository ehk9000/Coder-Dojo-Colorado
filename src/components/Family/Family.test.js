import React from 'react';
import { shallow } from 'enzyme';
import { Family } from './Family';

describe('Family', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Family />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
