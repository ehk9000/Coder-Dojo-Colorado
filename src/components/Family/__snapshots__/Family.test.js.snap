// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Family should match snapshot 1`] = `
<section
  className="Family"
>
  <div
    className="family-header"
  >
    <h2>
      Welcome, undefined!
    </h2>
  </div>
  <div
    className="family-main"
  >
    <div
      className="family-sidebar"
    >
      <h3>
        My Contact Info
      </h3>
      <h4>
        Phone: 
      </h4>
      <p />
      <h4>
        Email: 
      </h4>
      <p />
      <h4>
        Address: 
      </h4>
      <p>
        undefined, 
      </p>
      <p>
        undefined, undefined undefined
      </p>
    </div>
    <div
      className="family-details"
    >
      <h3
        className="students-header"
      >
        My Family
        <Link
          to="/myfamily/registerstudent"
        >
          <button
            type="button"
          >
            Add a Student
          </button>
        </Link>
      </h3>
      <div
        className="students-container"
      />
    </div>
  </div>
</section>
`;
